node_modules/col-activities/lib/api.js

```diff
@@ -1006,6 +1023,92 @@ var setImpactScore = function(score, assetId, callback) {
   });
 };
 
+/**
+ * Recalculate the trending scores for assets in one or all courses. Loop through all persisted
+ * activities in the course[s] that fall within the time interval specified in trendingScores.activityWindow
+ * and calculate an accurate total for each asset. If no impactful activities are found for an asset within 
+ * that time interval, its trending score is set to zero.
+ *
+ * @param  {Course}         [course]            The course for which the user points should be recalculated. If null,
+                                                recalculate for all courses.
+ * @param  {Function}       callback            Standard callback function
+ * @param  {Object}         callback.err        An error that occurred, if any
+ */
+var recalculateTrendingScores = module.exports.recalculateTrendingScores = function(course, callback) {
+  // Since impact scores can't be overriden, the default activities configuration suffices. Index by type
+  // for quick lookup.
+  var activityConfigurationByType = _.keyBy(ActivitiesDefaults, 'type');
+  // Get the activity window (in hours) specifying how far back we should query.
+  var activityWindow = config.get('trendingScores.activityWindow');
+  var cutoffDate = moment().subtract(activityWindow, 'hour').toDate();
+
+  // Get all recent asset-associated activities for the course[s]
+  var activityOptions = {
+    'where': {
+      'asset_id': {'$ne': null},
+      'created_at': {'gt': cutoffDate}
+    }
+  };
+
+  if (course) {
+    log.info({'course': course.id}, 'Will update trending scores for a single course');
+    activityOptions.where.course_id = course.id;
+  } else {
+    log.info('Will update trending scores for all courses');      
+  }
+  
+  DB.Activity.findAll(activityOptions).complete(function(err, activities) {
+    if (err) {
+      log.error({'err': err}, 'Failed to get activities for the course');
```

Comment:
@paulkerschen minor detail: Should error message above be sensitive to 'single course' vs. 'all courses'?